app/javascript/crayons/Pills/Pill.jsx

```diff
@@ -0,0 +1,93 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+import { useState } from 'preact/hooks';
+import classNames from 'classnames/bind';
+import { Icon } from '@crayons';
+import XIcon from '@images/x.svg';
+
+export const Pill = ({
+  children,
+  element = 'button',
+  iconLeft,
+  iconRight,
+  iconRightDestructive,
+  className,
+  tooltip,
+  onKeyUp,
+  ...otherProps
+}) => {
+  const Element = element;
+  const restOfProps =
+    element === 'button'
+      ? { type: 'button', onKeyUp: handleKeyUp }
+      : element === 'a'
+      ? { href: '#' }
+      : '';
```

Comment:
It's personal preference but I find nested ternaries like this pretty hard to read, and I'd have a preference for splitting it out even if it ends up being multiple lines.

I'm also a bit confused about use of links in this context. Their `href` is always hardcoded to `#` and we don't show any 'usual' link styles like the underline on hover/focus 🤔

Reply:
Re nested ternaries: how would you like this to be formatted? I have absolutely no opinion on this so I'm open :) 

Re that hardcoded `href` - whoops! Glad we have PR reviews :D

Reply:
You could use a switch/case. It's a little more verbose, but way easier to read.

```suggestion
let restOfProps;

switch (element) {
  case 'button':
    restOfProps = { type: 'button', onKeyUp: handleKeyUp };
    break;

  case 'a':
    restOfProps = { href: '#' };
    break;

  default:
    restOfProps = {};
}
```

Reply:
@nickytonline thank, that makes sense - i've updated it. cc @aitchiss